gems/pending/postgres_ha_admin/database_yml.rb

```diff
@@ -0,0 +1,59 @@
+require 'util/miq-password'
+require 'fileutils'
+
+module PostgresHaAdmin
+  class DatabaseYml
+    attr_reader :db_yml_file, :environment
+
+    def initialize(db_yml_file, environment)
+      @db_yml_file = db_yml_file
+      @environment = environment
+    end
+
+    def pg_params_from_database_yml
+      rails_params_to_pg(YAML.load_file(db_yml_file)[environment])
+    end
+
+    def update_database_yml(params)
+      db_yml = YAML.load_file(db_yml_file)
+      db_yml[environment].merge!(pg_parameters_to_rails(params))
+      remove_empty(db_yml[environment])
+
+      new_name = "#{db_yml_file}_#{Time.current.strftime("%d-%B-%Y_%H.%M.%S")}"
+      FileUtils.copy(db_yml_file, new_name)
+      begin
+        File.write(db_yml_file, db_yml.to_yaml)
+      rescue IOError => err
```

Comment:
Actually, I would rescue anything here. We really want to move the file back no matter what.